### PR TITLE
Add led and backlight support for yuga.

### PR DESCRIPTION
**Pull request by wouterfranken**
Add devicetree support for TI LM3533 LED driver.
Support LM3533 LED driver in Sony yuga DTS.

### CODE REVIEW

File: arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts [hunk 1]
```diff
@@ -373,6 +373,66 @@
...
 		};
 
+                gsbi@12440000 {
+                        status = "okay";
+                        qcom,mode = <GSBI_PROT_I2C>;
+
+                        i2c@12460000 {
+                                status = "okay";
+
+				clock-frequency = <200000>;
```
Comment (okias) [inline arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts:383]: fix formatting (tabs/spaces, it should align with original DTS)

File: arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts [hunk 2]
```diff
@@ -202,8 +202,8 @@
 				};
 
 				l11 {
-					regulator-min-microvolt = <3000000>;
-					regulator-max-microvolt = <3000000>;
+					regulator-min-microvolt = <2850000>;
```
Comment (okias) [inline arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts]: this should go as a separate change with explanation (I guess downstream doing it as u described here)?

File: drivers/mfd/lm3533-core.c [hunk 3]
```diff
@@ -14,13 +14,15 @@
 #include <linux/gpio.h>
 #include <linux/i2c.h>
 #include <linux/mfd/core.h>
+#include <linux/of_gpio.h>
```
Comment (okias) [inline drivers/mfd/lm3533-core.c:17]: can you do also YAML documentation for it in `Documentation/devicetree/bindings/` ?

Easiest way is copy existing binding for similar mfd and adjust it, then run yamllint, and `make dt_bindings_check`

**Review by okias — CHANGES_REQUESTED** [review]

File: Documentation/devicetree/bindings/mfd/ti,lm3533.yaml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+    i2c@12460000 {
+        compatible = "qcom,i2c-qup-v1.1.1";
+        ...
+
+        lm3533@36 {
+            compatible = "ti,lm3533";
+            reg = <0x36>;
+
+            hwen-gpios = <&pm8921_gpio 26 GPIO_ACTIVE_HIGH>;
```
Comment (okias) [inline Documentation/devicetree/bindings/mfd/ti,lm3533.yaml:158]: `GPIO_ACTIVE_HIGH` will need include for the example

File: Documentation/devicetree/bindings/mfd/ti,lm3533.yaml [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+additionalProperties: false
+
+examples:
+  - |
+    += EXAMPLE
+
+    i2c@12460000 {
+        compatible = "qcom,i2c-qup-v1.1.1";
+        ...
```
Comment (okias) [inline Documentation/devicetree/bindings/mfd/ti,lm3533.yaml]: not needed dots

File: Documentation/devicetree/bindings/mfd/ti,lm3533.yaml [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    required:
+      - reg
+      - label
+      - led-max-microamp
+
+additionalProperties: false
+
+examples:
+  - |
+    += EXAMPLE
```
Comment (okias) [inline Documentation/devicetree/bindings/mfd/ti,lm3533.yaml]: `make dt_bindings_check` will probably fails on this, remove

File: Documentation/devicetree/bindings/mfd/ti,lm3533.yaml [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    type: object
+
+    properties:
+      ti,als-resistance-ohms:
+        description: The resistor value (R_als).
+        minimum: 1574
+        maximum: 200000
+
+      ti,pwm-mode:
+        description:
```
Comment (okias) [inline Documentation/devicetree/bindings/mfd/ti,lm3533.yaml]: `>` or `|` missing (probably `>` is good choice)

... this is valid for all multiline description or other stuff

**Review by okias — CHANGES_REQUESTED** [review]

File: Documentation/devicetree/bindings/mfd/ti,lm3533.yaml (hunk 6 above)
Comment (wouterfranken) [reply Documentation/devicetree/bindings/mfd/ti,lm3533.yaml]: Strange, it did not. I will remove.

File: arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts (hunk 2 above)
Comment (wouterfranken) [reply arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts]: I put this in a separate commit. Should this be in a separate pull request?

Comment (okias) [reply arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts]: separation - that's just for upstreaming, for qcom-apq8064 tree, I don't care, but when you'll be sending `git send-patch`, it has to be separate, if it's not one change.

Don't forget some brief decription why you introducing the change (even for very small changes, at least one sentence of description is required)

Comment (wouterfranken) [reply arch/arm/boot/dts/qcom-apq8064-sony-xperia-yuga.dts]: Ok, opened another PR with regulator changes and a brief description to every commit.

**Comment by okias** [comment]
Thank you!

**Pull request merged**